homeassistant/components/switchbot_cloud/lock.py

```diff
@@ -51,3 +66,13 @@ async def async_unlock(self, **kwargs: Any) -> None:
         await self.send_api_command(LockCommands.UNLOCK)
         self._attr_is_locked = False
         self.async_write_ha_state()
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes."""
+        return {
+            ATTR_SW_VERSION: self._sw_version,
+            ATTR_BATTERY_LEVEL: self._battery_level,
+            ATTR_CALIBRATED: self._calibrated,
+            ATTR_DOOR_STATE: self._door_state,
+        }
```

Comment:
We should not have this as state attributes. the firmware version can be set on the device_info object. The battery level can be a sensor entity and calibrated and door state can be a binary sensor. These are more user friendly than just putting them in here